### PR TITLE
Add tests for the StaticRange constructor

### PR DESCRIPTION
**Pull request by sanketj**
A StaticRange constructor is being introduced in the WHATWG spec with this PR: https://github.com/whatwg/dom/pull/778.

Making a corresponding PR here to add tests for the same.

### CODE REVIEW

File: dom/ranges/StaticRange-constructor.html
Comment (rniwa) [inline dom/ranges/StaticRange-constructor.html]: This should be: `assert_equals(staticRange.startContainer, document, 'valid startContainer');`

**Review by rniwa — DISMISSED** [review]
Please fix the line 153 as I commented above.

**Comment by rniwa** [comment]
We also need to update imported/w3c/web-platform-tests/dom/interfaces.html which asserts that `StaticRange.length` is 0.

File: dom/ranges/StaticRange-constructor.html
Comment (sanketj) [reply dom/ranges/StaticRange-constructor.html]: Oops...fixed, thanks!

**Comment by sanketj** [comment]
> Please fix the line 153 as I commented above.

Fixed.

**Comment by sanketj** [comment]
> We also need to update dom/interfaces.html which asserts that `StaticRange.length` is 0.

@rniwa I don't see any entries for StaticRange in dom/interfaces.html. I must be missing something. Where exactly is the check for StaticRange.length?

**Comment by rniwa** [comment]
> > We also need to update dom/interfaces.html which asserts that `StaticRange.length` is 0.
> 
> @rniwa I don't see any entries for StaticRange in dom/interfaces.html. I must be missing something. Where exactly is the check for StaticRange.length?

It's autogenerated from https://github.com/web-platform-tests/wpt/blob/master/interfaces/dom.idl

**Comment by sanketj** [comment]
> > > We also need to update dom/interfaces.html which asserts that `StaticRange.length` is 0.
> > 
> > 
> > @rniwa I don't see any entries for StaticRange in dom/interfaces.html. I must be missing something. Where exactly is the check for StaticRange.length?
> 
> It's autogenerated from https://github.com/web-platform-tests/wpt/blob/master/interfaces/dom.idl

Ah, thanks! I've fixed that. Please take another look.

**Review by lukebjerring — APPROVED** [review]
Changes to IDL LGTM. Note that they'll automatically be scraped once the changes land in the spec, but I actually think it's good to preempt that with manually attaching the change to the test changes.

@foolip fyi in case you see a reversion PR created by reffy.

**Comment by foolip** [comment]
As long as https://github.com/whatwg/dom/pull/778 is merged at the same time, there will only be a short window for the reverting to happen, and the PR itself will warn that the file was recently modified manually. Let's not merge it and it'll be closed automatically once the spec change propagates.

**Comment by annevk** [comment]
@rniwa is this all okay now? I can land the specification change together with this. I think all is in order. Thanks for your work on this @sanketj, much appreciated!

**Comment by rniwa** [comment]
> @rniwa is this all okay now? I can land the specification change together with this. I think all is in order. Thanks for your work on this @sanketj, much appreciated!

Yup, the latest test is identical to [what I've checked into WebKit](https://trac.webkit.org/browser/webkit/trunk/LayoutTests/imported/w3c/web-platform-tests/dom/ranges/StaticRange-constructor.html) and [we pass all test cases](https://trac.webkit.org/browser/webkit/trunk/LayoutTests/imported/w3c/web-platform-tests/dom/ranges/StaticRange-constructor-expected.txt).

**Pull request merged**
